staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -259,8 +252,8 @@ func (d *DiscoveryClient) downloadLegacy() (
 
 	var resourcesByGV map[schema.GroupVersion]*metav1.APIResourceList
 	// Switch on content-type server responded with: aggregated or unaggregated.
-	switch responseContentType {
-	case AcceptV1:
+	switch {
+	case IsV1ContentType(responseContentType):
```

Comment:
before the aggregated discovery changes in the client, did we check content-type at all in the client? if not, we don't really know what load balancers or proxies were injecting into content-type, and I would consider reversing the order of this check, specifically recognizing the V2Beta1 discovery format, and falling back unconditionally to legacy v1 parsing, more like:

```go
switch {
case IsV2Beta1ContentType(responseContentType):
  ...
default:
  ... attempt to handle as v1
}
```

if we were checking content-type before, then narrowing IsV1ContentType to make sure the base content-type is correct and we don't have `as`/`g`/`v` params seems sufficient